src/core/reactor.cc

```diff
@@ -1800,10 +1797,7 @@ bool reactor::poll_any_event() {
         // actually performs work, but triggers no user continuations, so okay
         _backend->reap_kernel_completions() ||
         flush_pending_aio() ||
-#ifndef HAVE_OSV
-        _backend->kernel_submit_work() ||
-#endif
-        _backend->reap_kernel_completions());
+        _backend->submit_and_get_completions());
```

Comment:
Well this moves in a baby step, since we can't eliminate reap_kernel_completions as it has another call site.

I suppose it can be further merged later, this is already a big change.

Reply:
yeah, let's leave it for another change. so far, the goal is to push the implementation of poller down to `reactor_backend`. this will allow us to either customize them or further merge them.

but thinking forward, i am not quite if we can merge `reap_kernel_comletions()` with `submit_and_get_completions()`, as `flush_pending_aio()` is a oddball. it mutates `reactor::_io_queues`, which does not belong to `reactor_backend`. and we cannot reorder these three steps per the comment at https://github.com/scylladb/seastar/blob/17183ed4e475849e5c624c876cf37f6eabc64220/src/core/reactor.cc#L3186